tests/loop1.dora

```diff
@@ -1,5 +1,23 @@
 fun main() {
+  println(x().toString());
+  // println(y().toString()); // terminated by signal SIGTRAP
```

Comment:
This bug is extremely bad. It is very likely that this isn't related to your changes: Either it was already there or I might have broken it recently when removing exceptions...

Reply:
Agreed.

Reply:
Ok, this is actually not a bug on master but only in this PR. I am a bit relieved now, was losing confidence in the test suite... The problem is that `while` loops are not always returning a value. This test case shouldn't even compile.

Reply:
As mentioned a few lines below in the comment, loop has the same bug:

```
  // the loop keyword had the same bug:
  /*
  loop {
    break;
    return 5;
  }
  */
```

(I agree that this bug should be fixed, but I believe that's unrelated to this change.)

Reply:
Oh right, thanks! Nevertheless handling of `while` is not quite correct. AFAICS this bug is in `semck` and should be gone after removing `loop`, so I guess I am waiting until this PR is ready.

Reply:
> AFAICS this bug is in semck and should be gone after removing loop, so I guess I am waiting until this PR is ready.

Not quite – I moved the functionality that was formerly on `loop` (returning things) to `while`, such that `while true` is a replacement for `loop` in all cases, not just in some, see: https://github.com/dinfuehr/dora/pull/186/files#diff-08481ad25efc97149a0a31c6823d17cf

> I guess I am waiting until this PR is ready.

Mhh, I'd say it is ready – what remains to be done in this PR from your POV?

Reply:
I think you can remove that comment now.